src/test/run-pass/traits/trait-alias-objects.rs

```diff
@@ -21,7 +21,6 @@ pub fn main() {
     let b = Box::new(456) as Box<dyn Foo>;
     assert!(*b == 456);
 
-    // FIXME(alexreg): associated type should be gotten from trait alias definition
-    // let c: &dyn I32Iterator = &vec![123].into_iter();
-    // assert_eq!(c.next(), Some(123));
+    let c: &mut dyn I32Iterator<Item = u32> = &mut vec![123].into_iter();
```

Comment:
Should the commented out code also be legal?

Reply:
Actually *only* the commented-out code should be legal. I was messing around and forgot to revert. Thanks for the catch.